app/src/main/java/org/xbmc/kore/ui/sections/remote/RemoteActivity.java

```diff
@@ -389,6 +389,21 @@ protected void handleStartIntent(Intent intent, boolean queue) {
             videoUri = intent.getData();
         }
 
+        if (videoUri == null) {
+            // Some apps hide the link in the clip, try to detect any link by casting the intent
+            // to string a looking with a regular expression:
+
+            Matcher matcher = Pattern.compile("https?://[^\\s]+").matcher(intent.toString());
+            String matchedString = null;
+            if (matcher.find()) {
+                matchedString = matcher.group(0);
+                if (matchedString.endsWith("}")) {
+                    matchedString = matchedString.substring(0, matchedString.length() - 1);
+                }
+                videoUri = Uri.parse(matchedString);
+            }
+        }
```

Comment:
This whole `if (videoUri == null)` precedes the next one, which is similar, and deals with local sharing, so it can "capture" requests that should be done by it. I think that the logic should be unified, and that any division between the 2 logics should be explicitly acknowledge and solved.

Reply:
I've changed it to try to detect `videoUri` if `null`, then `toPluginUrl` will always be called. I have made some tests and it keeps working. Hope this makes sense.